### ISSUE TITLE
use of undeclared identifier 'MKMapTypeMutedStandard'

### ISSUE THREAD

**Issue by CptFabulouso** [react-community/react-native-maps#1900]
### Is this a bug report?
yes

### Have you read the [Installation Instructions](https://github.com/react-community/react-native-maps/blob/master/docs/installation.md)?
yes

### Environment
    react-native: 0.51.0
    react: 16.0.0
    react-native-maps: 0.19.0

    Target Platform: iOS - 10.3.1 Simulator (iPhone 5 - GoogleMaps)

### Steps to Reproduce
After the error below was happening in my current project, I tried creating brand new one 
`react-native init myApp`

installed react-native-maps
`npm install react-native-maps --save`

created pod file according to manual 
```
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'JobsApp' do
  rn_path = '../node_modules/react-native'

    pod 'yoga', path: "#{rn_path}/ReactCommon/yoga/yoga.podspec"
    pod 'React', path: rn_path, subspecs: [
      'Core',
      'RCTActionSheet',
      'RCTAnimation',
      'RCTGeolocation',
      'RCTImage',
      'RCTLinkingIOS',
      'RCTNetwork',
      'RCTSettings',
      'RCTText',
      'RCTVibration',
      'RCTWebSocket',
      'BatchedBridge'
    ]

    pod 'GoogleMaps'  # Remove this line if you don't want to support GoogleMaps on iOS
    pod 'react-native-maps', path: '../node_modules/react-native-maps'
    pod 'react-native-google-maps', path: '../node_modules/react-native-maps'  # Remove this line if you don't want to support GoogleMaps on iOS
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    if target.name == 'react-native-google-maps'
      target.build_configurations.each do |config|
        config.build_settings['CLANG_ENABLE_MODULES'] = 'No'
      end
    end
    if target.name == "React"
      target.remove_from_project
    end
  end
end
```
ran 
`pod install`

modified AppDelegate.m, final look:
```
/**
 * Copyright (c) 2015-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */

#import "AppDelegate.h"

#import <React/RCTBundleURLProvider.h>
#import <React/RCTRootView.h>

@import GoogleMaps;

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  NSURL *jsCodeLocation;

  jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];

  RCTRootView *rootView = [[RCTRootView alloc] initWithBundleURL:jsCodeLocation
                                                      moduleName:@"JobsApp"
                                               initialProperties:nil
                                                   launchOptions:launchOptions];
  rootView.backgroundColor = [[UIColor alloc] initWithRed:1.0f green:1.0f blue:1.0f alpha:1];

  [GMSServices provideAPIKey:@"MY_API_KEY"];
  
  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  UIViewController *rootViewController = [UIViewController new];
  rootViewController.view = rootView;
  self.window.rootViewController = rootViewController;
  [self.window makeKeyAndVisible];
  return YES;
}

@end
```

### Expected Behavior
compile and run

### Actual Behavior
failed during compilation due to this error
<img width="1000" alt="snimek obrazovky 2017-12-18 v 20 22 40" src="https://user-images.githubusercontent.com/25827002/34123927-6e7d5498-e431-11e7-8b37-ceac8c8b58b7.png">

the error in log
<img width="730" alt="snimek obrazovky 2017-12-18 v 20 27 28" src="https://user-images.githubusercontent.com/25827002/34124079-ea12f158-e431-11e7-9bb5-2c622cb8f705.png">

If I change or remove the identifier as xcode suggests, other errors pop up
**EDIT:** other errors do not pop up If I change or remove the identifier, it was in another project, which had more problems

**Comment by rborn** [comment]
@CptFabulouso  It looks to me that you are trying to use google maps with a map type that exists only in apple maps  - https://github.com/react-community/react-native-maps/blob/master/docs/mapview.md - search for `mapType`

**Comment by CptFabulouso** [comment]
The project fails during compilation (build). I did not touch the RN project and I am not specifying any mapType anywhere anyway, it would just render the default welcome screen. I've read a lot about setting search paths in `Build Settings`, so can a problem be there maybe? I really have no idea what's happening in xCode.

Anyway, maybe they will help:
`Framework Search Paths`
<img width="510" alt="snimek obrazovky 2017-12-19 v 0 21 59" src="https://user-images.githubusercontent.com/25827002/34133048-08c8252a-e453-11e7-9c8d-70d1c16edc22.png">

`Header Search Paths`
<img width="522" alt="snimek obrazovky 2017-12-19 v 0 22 06" src="https://user-images.githubusercontent.com/25827002/34133063-1830f2bc-e453-11e7-9539-ad7083e9a0f5.png">

Also my `Link Binary with Libraries`
<img width="696" alt="snimek obrazovky 2017-12-19 v 0 22 14" src="https://user-images.githubusercontent.com/25827002/34133085-36fd409c-e453-11e7-9e9a-f3021e45cd55.png">

**Comment by alvelig** [comment]
@CptFabulouso does it build if you close xCode and run react-native run-ios?
I just followed your steps and could not reproduce your error :(

**Comment by alvelig** [comment]
@rborn It should be underlined in the doc that `.xcworkspace` should be opened (I know that's lame but just in case)
***BOLD H1*** LOL

**Comment by CptFabulouso** [comment]
`react-native run-ios` does not build either. 
It ends with error
```
The following build commands failed:

	CompileC /...path/JobsApp/ios/build/Build/Intermediates/Pods.build/Debug-iphonesimulator/react-native-maps.build/Objects-normal/i386/RCTConvert+AirMap.o 
/...path/JobsApp/node_modules/react-native-maps/lib/ios/AirMaps/RCTConvert+AirMap.m normal i386 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)

Installing build/Build/Products/Debug-iphonesimulator/JobsApp.app
An error was encountered processing the command (domain=NSPOSIXErrorDomain, code=2):
Failed to install the requested application
An application bundle was not found at the provided path.
Provide a valid path to the desired application bundle.
Print: Entry, ":CFBundleIdentifier", Does Not Exist

Command failed: /usr/libexec/PlistBuddy -c Print:CFBundleIdentifier build/Build/Products/Debug-iphonesimulator/JobsApp.app/Info.plist
Print: Entry, ":CFBundleIdentifier", Does Not Exist
```
so there is a failure at the same location as in xCode (at RCTConvert+AirMap.m), but it seems it still tries to install the app anyway, but another failure with CFBundleIdentifier happens. There was some surprising mess with the CFBundleIdentifier, which I tried to fix, but no luck... or maybe it's just consequence of the first failure, dunno. I will try it again tomorrow on yet another new project.

and yes, I am opening the `.xcworkspace`, I've come across cocoapods before unfortunately, it's never a good sign.

**Comment by CptFabulouso** [comment]
So this time I created and setup project like above without even touching xCode and still got the same error. One thing that happens during `pod install` is some weird logout (at the end) with duplicate UUIDs. Don't know if that is normal

```
Analyzing dependencies
Fetching podspec for `React` from `../node_modules/react-native`
Fetching podspec for `react-native-google-maps` from `../node_modules/react-native-maps`
Fetching podspec for `react-native-maps` from `../node_modules/react-native-maps`
Fetching podspec for `yoga` from `../node_modules/react-native/ReactCommon/yoga/yoga.podspec`
Downloading dependencies
Installing GoogleMaps (2.5.0)
Installing React (0.51.0)
Installing react-native-google-maps (0.19.0)
Installing react-native-maps (0.19.0)
Installing yoga (0.51.0.React)
Generating Pods project
Integrating client project

[!] Please close any current Xcode sessions and use `maptest.xcworkspace` for this project from now on.
Sending stats
Pod installation complete! There are 16 dependencies from the Podfile and 5 total pods installed.

[!] Automatically assigning platform ios with version 8.0 on target maptest because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.

[!] [Xcodeproj] Generated duplicate UUIDs:

XCBuildConfiguration -- 
XCBuildConfiguration -- 
PBXBuildFile -- 
PBXBuildFile -- 
//...this logs out many times...
PBXBuildFile -- 
PBXBuildFile -- 
PBXContainerItemProxy -- 
```

**Comment by CptFabulouso** [comment]
One thing that is different is that it does build after removing the error line 
`@"mutedStandard": @(MKMapTypeMutedStandard)` from the file `RCTConvert+AirMap.m`

**Comment by alvelig** [comment]
Hm, I saw this recently (referenced above). 

What XCode version are you using? MKMapTypeMutedStandard was introduced for iOS 11. That would make sense if you have XCode 8.3.

**Comment by martyjg** [comment]
@CptFabulouso 

When I followed Xcode's suggestion to change the identifier for @"mutedStandard" to @(MKMapTypeStandard) and then built and ran from Xcode it worked fine.

What kind of other errors did it produce?

**Comment by alvelig** [comment]
@CptFabulouso @martyjg what version of XCode do you have? 8.3 or 9?

**Comment by martyjg** [comment]
Version 8.3.3

**Comment by alvelig** [comment]
Yep, that explains: 
> MKMapTypeMutedStandard was introduced for iOS 11.
Great, thank you all. I think we've found the problem 👍

**Comment by alvelig** [comment]
@rborn it think that also would be welcome to the docs update :) To warn to update XCode to 9.+

**Comment by CptFabulouso** [comment]
Yes, I use version 8.3.3 too.

@martyjg I assume you are referring to this line in my original post

> If I change or remove the identifier as xcode suggests, other errors pop up

I think they popped up just in the original project (which I simply abandoned, it's not for production, just for learning). But because I was receiving the same `MKMapTypeMutedStandard` error in the test project described here, I assumed changing the line would lead to the same other errors, but didn't as I found later. In the original project, XCode was also unable to find GoogleMaps in AppDelegate, so there were some configuration problems. I will edit the issue description.
